ts/utils/dates.ts

```diff
@@ -1,13 +1,41 @@
-import { format } from "date-fns";
+import { format as dateFnsFormat } from "date-fns";
+import en from "date-fns/locale/en";
+import it from "date-fns/locale/it";
+import I18n from "../i18n";
+
+type Locales = {
+  [index: string]: object;
+};
+
+const locales: Locales = { it, en };
```

Comment:
reuse [this](https://github.com/teamdigitale/io-app/blob/master/ts/i18n.ts#L13)

Reply:
date-fns library lets you specify in the format function the locale you want the date to be translated to. This parameter is an object coming from this library itself, having inner methods like:
```
module.exports = {
  distanceInWords: buildDistanceInWordsLocale(),
  format: buildFormatLocale()
}
```
 so I need to use this and not i18n.ts supported translations

Reply:
what I mean is that we can't list the locales (it, en) in two different places (I18N and here), there need to be only one place where the list of supported locales is defined - I don't have a solution to suggest right now - cc @fpersico

Reply:
@cloudify What you think about this?

```
import { format as dateFnsFormat } from "date-fns";
import dfns_en from "date-fns/locale/en";
import dfns_it from "date-fns/locale/it";
import I18n from "../i18n";
import { Locales } from "../../locales/locales";

type DFNSLocales = { [index in Locales]: object };

const locales: DFNSLocales = { it: dfns_it, en: dfns_en };
```

Reply:
No way to add Langs not in I18n locales and force you to add a date-fns locale for each i18n one.

Reply:
doesn't solve the issue - if we add a new locale, `locales` is still valid, `[index in Locales]` just says it should be a valid locale, it doesn't guarantee we include all locales

Reply:
what I want is to avoid having to remember to add a locale in this file too

Reply:
@cloudify [index in "A" | "B"] means that object MUST have both A and B.

https://www.typescriptlang.org/play/#src=type%20X%20%3D%20%7B%0D%0A%20%20%20%20%5Bindex%20in%20%22A%22%20%7C%20%22B%22%5D%3A%20number%0D%0A%7D%0D%0A%0D%0Aconst%20obj%3A%20X%20%3D%20%7B%0D%0A%20%20%20%20A%3A%205%2C%0D%0A%7D

Reply:
TIL! so this is our solution :) thanks!

Reply:
changed